mlx_lm/models/cache.py

```diff
@@ -659,6 +659,55 @@ def extend(self, other):
             c.extend(other)
 
 
+class KCache(_BaseCache):
```

Comment:
Maybe it makes sense to make `KVCache` general to an arbitrary number of arrays?